tests/integration/test_apt.py

```diff
@@ -44,53 +44,147 @@ def test_install_package_error():
 def test_remove_package():
     # First ensure the package is present
     cfssl = apt.DebianPackage.from_apt_cache("golang-cfssl")
-    if not cfssl.present:
-        apt.add_package("golang-cfssl")
-    assert get_command_path("cfssl") == "/usr/bin/cfssl"
+    assert not cfssl.present
+    # Add package
+    apt.add_package("golang-cfssl")
+    assert get_command_path("cfssl")
+    subprocess.run(["cfssl", "version"], check=True)
     # Now remove the package and check its bins disappear too
     apt.remove_package("golang-cfssl")
-    assert get_command_path("cfssl") == ""
-
-
-def test_install_package_external_repository():
-    repositories = apt.RepositoryMapping()
-
-    # Get the Hashicorp GPG key
-    key = urlopen("https://apt.releases.hashicorp.com/gpg").read().decode()
-
-    # Add the hashicorp repository if it doesn't already exist
-    if "deb-apt.releases.hashicorp.com-focal" not in repositories:
-        line = "deb [arch=amd64] https://apt.releases.hashicorp.com focal main"
-        repo = apt.DebianRepository.from_repo_line(line)
-        # Import the repository's key
-        repo.import_key(key)
-        repositories.add(repo)
-
+    assert not get_command_path("cfssl")
+
+
+def test_install_package_from_external_repository():
+    repo_id = "deb-https://repo.mongodb.org/apt/ubuntu-jammy/mongodb-org/8.0"
+    repos = apt.RepositoryMapping()
+    assert repo_id not in repos
+    assert not get_command_path("mongod")
+    ## steps
+    key = urlopen("https://www.mongodb.org/static/pgp/server-8.0.asc").read().decode()
+    key_file = apt.import_key(key)
+    line = (
+        "deb [ arch=amd64,arm64 signed-by={} ]"
+        " https://repo.mongodb.org/apt/ubuntu jammy/mongodb-org/8.0 multiverse"
+    ).format(key_file)
+    ## if: use implicit write_file
+    repo = apt.DebianRepository.from_repo_line(line)  # write_file=True by default
+    assert repo_id not in repos
+    ## if: don't use implicit write_file
+    # repo = apt.DebianRepository.from_repo_line(line, write_file=False)
+    # repos.add(repo)
+    # assert repo_id in repos
+    ## fi
+    assert repo_id in apt.RepositoryMapping()
     apt.update()
-    apt.add_package("terraform")
-
-    assert get_command_path("terraform") == "/usr/local/bin/terraform"
+    apt.add_package("mongodb-org")
+    assert get_command_path("mongod")
+    subprocess.run(["mongod", "--version"], check=True)
+    ## cleanup
+    os.remove(key_file)
+    apt._add_repository(repo, remove=True)  # pyright: ignore[reportPrivateUsage]
+    assert repo_id not in apt.RepositoryMapping()
+    apt.update()
+    apt.remove_package("mongodb-org")
+    assert get_command_path("mongod")  # mongodb-org is a metapackage
+    subprocess.run(["apt", "autoremove"], check=True)
+    assert not get_command_path("mongod")
+
+
+def test_install_higher_version_package_from_external_repository():
+    repo_id = "deb-https://ppa.launchpadcontent.net/inkscape.dev/stable/ubuntu/-jammy"
```

Comment:
This test is kinda slow. Maybe better to pull the version out of the package listing without installing it?

Reply:
I wouldn't be too worried if the integration tests are only 5 minutes. Or maybe we can find a lighter package to install? If you can't find one fairly quickly, don't worry about it.